files/common/var/lib/delphix-platform/ansible/10-delphix-platform/roles/delphix-platform/tasks/main.yml

```diff
@@ -718,3 +718,12 @@
     name: "nullmailer"
     state: "stopped"
   when: not ansible_is_chroot
+
+#
+# Ensure /snap/bin is removed from PATH in /etc/environment
+#
+- lineinfile:
+    path: /etc/environment
+    regexp: '^PATH='
+    line: "{{ lookup('file', '/etc/environment') | regex_search('^PATH=.*') | regex_replace(':/snap/bin', '') }}"
```

Comment:
It looks like the `PATH` setting in `/etc/environment` comes from the `libpam-modules` package's postinst script (`/var/lib/dpkg/info/libpam-modules:amd64.postinst`):
```
# Add PATH to /etc/environment if it's not present there or in
# /etc/security/pam_env.conf
if [ "$1" = "configure" ] && dpkg --compare-versions "$2" lt 1.3.1-5ubuntu4.1; then
        if ! grep -qs ^PATH /etc/security/pam_env.conf; then
                if ! grep -qs ^PATH= /etc/environment; then
                        echo 'PATH="/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin"' >> /etc/environment
                elif ! grep -qs "^PATH=.*/snap/bin" /etc/environment; then
                        sed -i '/^PATH="/ s,"$,:/snap/bin",g' /etc/environment
                fi
        fi
fi
```
That's pretty interesting, so this script does not set a default `PATH` if there is already one configured in `/etc/security/pam_env.conf`... Perhaps we should consider simply setting the default `PATH` exactly how we want it in there instead...

Reply:
Yes, that's a great idea. Thanks @sebroy for pointing this out. 

Just one point: in the case of an upgrade, if we set this in `/etc/security/pam_env.conf`, the global PATH will refer to `pam_env.conf` only, but the PATH will still be defined in the `/etc/environment` file with some garbage values. Is that okay?

Reply:
Good point, we could remove the PATH line from `/etc/environment` in an adjacent part of the ansible task.

Reply:
Yeah updated, can you please re-review?